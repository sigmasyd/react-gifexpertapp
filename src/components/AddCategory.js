import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setsInputValue] = useState('');

  const handleInputChange = (e) => {
    //console.log(e.target.value);
    setsInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    console.log('Submit hecho');
    if (inputValue.trim().length > 2) {
      setCategories((cat) => [inputValue, ...cat]);
      setsInputValue('');
    }
  };
  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
    </form>
  );
};
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
