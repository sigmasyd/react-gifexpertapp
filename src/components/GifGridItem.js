import React from 'react';

export const GifGridItem = ({ title, url }) => {
  //console.log(title, url);
  return (
    <div className="card animate__animated animate__fadeInLeft">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};
