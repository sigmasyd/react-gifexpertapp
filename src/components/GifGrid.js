import React, { useEffect, useState } from 'react';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
  const [images, setImages] = useState([]);

  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      'https://api.giphy.com/v1/gifs/search?api_key=anyoG6FgTEQ3Fx2KywvgvcZGe1b9Bv1Y&q=sakura&limit=10';
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((img) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      };
    });

    console.log(gifs);
    setImages(gifs);
  };

  return (
    <>
      <h3>{category}</h3>
      {images.map((img) => (
        <GifGridItem key={img.id} {...img} />
      ))}
    </>
  );
};
