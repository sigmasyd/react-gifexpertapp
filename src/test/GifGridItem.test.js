import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../components/GifGridItem';

describe('Pruebas sobre el componente GifFridItem', () => {
  test('debe de mostrar el componente correctamente', () => {
    const title = 'pokemon';
    const url = 'https://image.com';
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    expect(wrapper).toMatchSnapshot();
  });
});
