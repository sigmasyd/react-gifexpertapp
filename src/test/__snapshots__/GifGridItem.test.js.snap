// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifFridItem debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeInLeft"
>
  <img
    alt="pokemon"
    src="https://image.com"
  />
  <p>
    pokemon
  </p>
</div>
`;
